### ISSUE TITLE
Failed to create server implementation. Please report this to the FoliaLib GitHub issues page. Forks of server software may not all be supported. If you are using an unofficial fork, please report this to the fork's developers first.

### ISSUE THREAD

**Issue by nennneko5787** [TechnicallyCoded/FoliaLib#15]
It was displayed when LunaChat was made compatible with Folia.
Here is the code: https://github.com/Earth2B2TMC/LunaChat/blob/master/src/main/java/com/github/ucchyocean/lc3/LunaChatBukkit.java
```
[11:52:53 ERROR]: Error occurred while enabling LunaChat v3.0.16 (Is it up to date?)
java.lang.IllegalStateException: Failed to create server implementation. Please report this to the FoliaLib GitHub issues page. Forks of server software may not all be supported. If you are using an unofficial fork, please report this to the fork's developers first.
        at com.tcoded.folialib.FoliaLib.<init>(FoliaLib.java:38) ~[LunaChat.jar:?]
        at com.github.ucchyocean.lc3.LunaChatBukkit.onEnable(LunaChatBukkit.java:77) ~[LunaChat.jar:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:287) ~[folia-api-1.20.4-R0.1-SNAPSHOT.jar:?]
        at io.papermc.paper.plugin.manager.PaperPluginInstanceManager.enablePlugin(PaperPluginInstanceManager.java:188) ~[folia-1.20.4.jar:git-Folia-"25ee657"]
        at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.enablePlugin(PaperPluginManagerImpl.java:104) ~[folia-1.20.4.jar:git-Folia-"25ee657"]
        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:507) ~[folia-api-1.20.4-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.craftbukkit.v1_20_R3.CraftServer.enablePlugin(CraftServer.java:645) ~[folia-1.20.4.jar:git-Folia-"25ee657"]
        at org.bukkit.craftbukkit.v1_20_R3.CraftServer.enablePlugins(CraftServer.java:556) ~[folia-1.20.4.jar:git-Folia-"25ee657"]
        at net.minecraft.server.MinecraftServer.loadWorld0(MinecraftServer.java:713) ~[folia-1.20.4.jar:git-Folia-"25ee657"]
        at net.minecraft.server.MinecraftServer.loadLevel(MinecraftServer.java:458) ~[folia-1.20.4.jar:git-Folia-"25ee657"]
        at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:309) ~[folia-1.20.4.jar:git-Folia-"25ee657"]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1217) ~[folia-1.20.4.jar:git-Folia-"25ee657"]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:346) ~[folia-1.20.4.jar:git-Folia-"25ee657"]
        at java.lang.Thread.run(Thread.java:840) ~[?:?]
```

**Comment by nennneko5787** [comment]
Result of executing the about command:
```
about
[11:59:51 INFO]: Checking version, please wait...
[11:59:52 INFO]: This server is running Folia version git-Folia-"25ee657" (MC: 1.20.4) (Implementing API version 1.20.4-R0.1-SNAPSHOT) (Git: 25ee657)
You are running the latest version
```

**Comment by nennneko5787** [comment]
![image](https://github.com/TechnicallyCoded/FoliaLib/assets/95109758/f61ef4b7-0b5c-47cb-b028-0f3901f2a069)
It is supposed to return absolutely null, what does this mean?

**Comment by nennneko5787** [comment]
#13 の複製かもしれん
まじでごめんなさい

**Comment by TechnicallyCoded** [comment]
Hey, @nennneko5787 I was probably very busy when you posted this issue a few months ago, so I unfortunately missed it. According to what you wrote above, translated by Google:

```
This may be a duplicate of the above, I'm really sorry
```

It does seem that your issue is exactly what was already solved in #13 . Make sure you relocate folialib and force the shade plugin to keep the entire FoliaLib library. I don't have any unnecessary bloat classes in the lib so the minimization process won't help on FoliaLib anyway :)
